File: Java/3228. Maximum Number of Operations to Move Ones to the End.java
class Solution {
    public int maxOperations(String s) {
        int i = 0, ans = 0, streak = 0;
        while (i < s.length()) {
            if (s.charAt(i) == '1') {
                streak++;
                i++;
            } else {
                while (i < s.length() && s.charAt(i) == '0') {
                    i++;
                }
                ans += streak;
            }
        }
        return ans;
    }
}

File: Python3/3228. Maximum Number of Operations to Move Ones to the End.py
class Solution:
    def maxOperations(self, s: str) -> int:
        ans = streak = i = 0

        while i < len(s):
            if s[i] == '1':
                streak += 1
                i += 1
            else:
                while i < len(s) and s[i] == '0':
                    i += 1
                ans += streak

        return ans